tests/src/unit-conversions.cpp

```diff
@@ -1657,6 +1657,94 @@ TEST_CASE("JSON to enum mapping")
     }
 }
 
+#ifdef __cpp_exceptions
+    #undef __cpp_exceptions
+    #define __cpp_exceptions 1
+#endif
+
+#ifdef JSON_NOEXCEPTION
+    #define JSON_NOEXCEPTION 0
+#endif
+
+#if (defined(__cpp_exceptions) || defined(__EXCEPTIONS) || defined(_CPPUNWIND)) && !defined(JSON_NOEXCEPTION)
```

Comment:
for some reason, if I don't have this defined I get the following compilation errors
![image](https://github.com/user-attachments/assets/7515b6e3-6222-4a0c-a23c-298005612223)